src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Parses the user input.
+ */
+public class Parser {
+
+    /**
+     * Creates todo when given userInput.
+     * @param userInput the input by the user.
+     * @return the todo inputted by the user.
+     * @throws DukeException if no description is inputted.
+     */
+    private static Todo createTodo(String userInput) throws DukeException {
+        String description = userInput.substring(4).trim();
+        if (description.length() == 0) {
+            throw new DukeException(":( OOPS!!! The description cannot be empty.");
+        }
+        return new Todo(description);
+    }
+
+    /**
+     * Creates deadline when given userInput.
+     * @param userInput the input by the user.
+     * @return the deadline inputted by the user.
+     * @throws DukeException if no date and time or description of deadline is inputted.
+     */
+    private static Deadline createDeadline(String userInput) throws DukeException {
+        String[] splitByBy = userInput.split(" /by ");
```

Comment:
is there a reason why splitByBy and not simply splitBy? thinks it works the same as well, splitByBy sounds kinda funky haha